hack/README.md

```
@@ -22,3 +22,40 @@ Example usage:
 ```shell
 ./hack/release-list-jiras.sh v1.2.35 v1.2.34
 ```
+
+## changelog-generate.sh
+
+This helper generates the historical `CHANGELOG.md` content using
+[`git-cliff`](https://git-cliff.org/) and the repository `cliff.toml` config.
+
+Generate the full historical changelog from all stable tags:
+
+```shell
```

markdownlint-cli2 (0.22.1)

[warning] Code block style
Expected: indented; Actual: fenced

(MD046, code-block-style)